torch/_inductor/codegen/cpp.py

```diff
@@ -3235,6 +3178,116 @@ def select_tiling(
+                                    stride = _try_get_stride(
+                                        index, tiling_factor, tiling_indices
+                                    )
+                                    if stride is not None and stride not in [0, 1]:
+                                        _update_negative_op_count(
+                                            _node.target, negative_op_type_count
+                                        )
+                            elif _node.target == "store":
+                                index = sub_block.body.indexing_exprs[
+                                    _node.args[2].args[0]
+                                ]
+                                if len(tiling_indices) == 1:
+                                    stride = _try_get_stride(
+                                        index, tiling_factor, tiling_indices
+                                    )
+                                    if stride is not None and stride != 1:
+                                        _update_negative_op_count(
+                                            _node.target, negative_op_type_count
+                                        )
+
+                            if isinstance(_node.target, str) and not (
+                                _node.target.startswith("masked_subblock")
+                                or _node.target
+                                in ["ops", "output", "constant", "get_index"]
+                            ):
+                                if _node.target not in op_type_count:
+                                    op_type_count[_node.target] = 1
+                                else:
+                                    op_type_count[_node.target] += 1
+
+                total_op_count = sum(op_type_count.values())
+                negative_op_count = sum(negative_op_type_count.values())
+                threshold = 0.03
+                if (
+                    total_op_count > 0
+                    and negative_op_count / total_op_count >= threshold
+                ):
+                    # Too many non-contiguous load/store/index_expr which hurts the
+                    # vectorization performance. Disable vectorization when exceeding
+                    # the threshold.
+                    return [], []
+
+                if (
+                    not reduction_group
+                    and group
+                    and not has_free_symbols(group[-1])
+                    and group[-1] < tiling_factor / 2
+                ):
+                    # For case of Multi Thread AMP Static shape of pyhpc_isoneutral_mixing,
+                    # the inner dim doesn't have enough elements to do vectorization
+                    # explicitly. Found that `#pragma GCC ivdep` has better performance than
+                    # `#pragma omp simd simdlen(8)`. Disable vectorization for this case.
+                    return [], []
```

Comment:
I'm wondering if it makes sense to always disable vectorization if the range of the loop is less than the vector length and in the scalar kernel, we enable things like `#pragma omp simd simdlen(8)` on demand?

Reply:
Thanks for the comment. I have added a TODO for this. How about we test the current heuristics among the 3 suites first? If everything looks good, we can proceed with trying this. I will update the test results here once done.

Reply:
Sure, sounds good.